workflow-templates/im-deploy-windows-service.yml

```
@@ -0,0 +1,396 @@
+      PAGERDUTY_WINDOW_DESC: 'Deploying Code to ${{ needs.set-vars.outputs.ENVIRONMENT }} from GitHub Actions' # TODO:  Verify this PD Maintenance Window Description
+      ASSET_ZIP: 'published_app.zip' # TODO: Verify that this wasn't changed in the CI build.  This is the value in that workflow by default.
+      UNZIPPED_ASSET: 'published_app'
+      DEPLOY_ZIP: 'deploy.zip'
+
+    steps:
+      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}"
+
+      - name: Open a PagerDuty Maintenance Window
+        id: open-window
+        uses: im-open/open-pagerduty-maintenance-window@v1.0.1
+        with:
+          pagerduty-api-key: ${{ secrets.PAGERDUTY_API_KEY }} # This is an env-level secret
+          description: '${{ env.PAGERDUTY_WINDOW_DESC }}'
+          minutes: ${{ env.PAGERDUTY_WINDOW_IN_MIN }}
+          service-id: ${{ secrets.PAGERDUTY_SERVICE_ID }} # This is an env-level secret
+
+      - run: |
+          echo "The maintenance window ID is: ${{ steps.open-window.outputs.maintenance-window-id }}"
+
+      - name: Download artifacts from release
+        uses: im-open/download-release-asset@v1.0.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }} # Special per-job token generated by GH for interacting with the repo
+          asset-name: ${{ env.ASSET_ZIP }}
+          tag-name: ${{ github.event.inputs.tag}}
+
+      - name: Unzip release asset
+        run: unzip -qq ${{ env.ASSET_ZIP }} -d ./${{ env.UNZIPPED_ASSET }}
+
+      # TODO:  If you need to do replacements in your appsettings.json file you can do it two ways.
+      # 1- Repeat this add-or-update-json-properties action for each environment, and add an 'if: needs.set-vars.outputs.ENVIRONMENT == ""' condition with the appropriate env to each action.
+      # 2- In the set-vars job, add one set-variable-based-on-environment action block per variable then use that variable in the single add-or-update-json-properties action below.
+      # - name: Update appsettings.json with Environment Specific Values
+      #   uses: im-open/add-or-update-json-properties@v1.0.1
+      #   with:
+      #     path-to-json-file: '${{ env.UNZIPPED_ASSET }}/appsettings.json'
+      #     properties-to-update-or-add: |
+      #       [
+      #         {"SecretName": "SecretValue"},
+      #         {"Nested.Secret.Name": "SecretValue"},
+      #         {"": ""}
+      #       ]
+
+      # TODO:  If you need to do any Octopus variable substitution (i.e. replacing #{OCTO_PLACEHOLDER} in files) use the following action.  Must be done once per file.
+      # - uses: im-open/octostache-action@v1.0.0
+      #   with:
+      #     variablesFile: '' # Files containing the variable substitutions to make, you will need to create this file
+      #     templateFile: ''  # The file to make the substitutions in
+      #     outputFile: ''    # The file to save the substitutions in (generally the same as above)
+
```

This is the new block that's been added to the other deployment workflows.
```suggestion
      # TODO:  If you need any json, yml or xml file (web.config/app.config/nlog.config) substitutions use the following
      #        action, otherwise delete it. This action won't add or remove items, it will just update the values.
      - uses: microsoft/variable-substitution@v1
        with:
          # TODO: add a comma separated list of files and the patterns, like './We*.config, ./Nlog.config'
          # The unzipped directory will not match your original solution structure.  The zip file only contains the published files for the project.
          files: ''
        env:
          # TODO:  replace examples with actual substitutions
          Value1: 'Value1' # Example showing replacement at a root level node
          Nested.Value.Name: 'Value2' # Example showing replacement at a nested node
          SecretValue: ${{ secrets.VALUE }} # Example showing replacement with a secret
```